### PR TITLE
Add Simple GPU Test: CUDA-VECTOR-ADD for aws-k8s-tester eks

### PR DESCRIPTION
**Pull request by PatrickXYS**
*Description of changes:*
Add a simple GPU test, which utilize gpu resources to execute vector-add operation. 

*Test:*
Tested locally on my dev-stack.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by PatrickXYS** [comment]
@gyuho Can you take a look?

File: eksconfig/add-on-cuda-vector-add.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
+package eksconfig
+
+import (
+	"github.com/aws/aws-k8s-tester/pkg/timeutil"
+)
+
+// AddOnCudaVectorAdd defines parameters for EKS cluster
+// add-on Cuda Vector Add.
+
```
Comment (gyuho) [inline eksconfig/add-on-cuda-vector-add.go:9]: Please remove this line.

File: eks/cuda-vector-add/cuda-vector-add.go [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+
+		ts.cfg.Logger.Info("querying cuda-vector-add")
+		ctx, cancel := context.WithTimeout(context.Background(), 15*time.Second)
+		out, err := ts.cfg.K8SClient.
+			KubernetesClientSet().
+			CoreV1().
+			Pods(ts.cfg.EKSConfig.AddOnCudaVectorAdd.Namespace).
+			Get(
+				ctx,
+				"vector-add",
```
Comment (gyuho) [inline eks/cuda-vector-add/cuda-vector-add.go:143]: Can we define `const` for this pod name?

File: eks/cuda-vector-add/cuda-vector-add.go [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+				metav1.GetOptions{},
+				)
+		cancel()
+
```
Comment (gyuho) [inline eks/cuda-vector-add/cuda-vector-add.go:147]: Remove line empty line.

File: eks/cuda-vector-add/cuda-vector-add.go [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+	if err := k8s_client.CreateNamespace(
+		ts.cfg.Logger,
+		ts.cfg.K8SClient.KubernetesClientSet(),
+		ts.cfg.EKSConfig.AddOnCudaVectorAdd.Namespace,
+	); err != nil {
+		return err
+	}
+
+	ts.cfg.Logger.Info("creating pod",
+		zap.String("name", "CudaVectorAdd"),
```
Comment (gyuho) [inline eks/cuda-vector-add/cuda-vector-add.go:85]: can we define pod name as "const"

File: eks/cuda-vector-add/cuda-vector-add.go [hunk 5]
```diff
@@ -0,0 +1,231 @@
+package cuda_vector_add
+
+import (
+	"fmt"
+	eks_tester "github.com/aws/aws-k8s-tester/eks/tester"
```
Comment (gyuho) [inline eks/cuda-vector-add/cuda-vector-add.go:5]: let's move all non-standard packages below, and run `gofmt` (`./scripts/fmt.sh`)

File: eksconfig/add-on-cuda-vector-add.go (hunk 1 above)
Comment (PatrickXYS) [reply eksconfig/add-on-cuda-vector-add.go:9]: Addressed in the new commit.

File: eks/cuda-vector-add/cuda-vector-add.go (hunk 2 above)
Comment (PatrickXYS) [reply eks/cuda-vector-add/cuda-vector-add.go:143]: Yeah good point, fixed.

File: eks/cuda-vector-add/cuda-vector-add.go (hunk 4 above)
Comment (PatrickXYS) [reply eks/cuda-vector-add/cuda-vector-add.go:85]: Yeah good point, fixed.

File: eks/cuda-vector-add/cuda-vector-add.go (hunk 3 above)
Comment (PatrickXYS) [reply eks/cuda-vector-add/cuda-vector-add.go:147]: Fixed, btw, is there any Golang style/format doc that our team should follow?

File: eks/cuda-vector-add/cuda-vector-add.go (hunk 4 above)
Comment (PatrickXYS) [reply eks/cuda-vector-add/cuda-vector-add.go:85]: Yeah good point, fixed.

File: eks/cuda-vector-add/cuda-vector-add.go (hunk 5 above)
Comment (PatrickXYS) [reply eks/cuda-vector-add/cuda-vector-add.go:5]: Fixed.

File: eks/cuda-vector-add/cuda-vector-add.go [hunk 6]
```diff
@@ -0,0 +1,254 @@
...
+
+$ kubectl apply -f vector-add.yaml
+pod "vector-add" created
+
+$ kubectl get pods/vector-add
+NAME                   READY     STATUS      RESTARTS   AGE
+vector-add       0/1       Completed   0     4s
+*/
+
+func (ts *tester) Delete() error {
```
Comment (gyuho) [inline eks/cuda-vector-add/cuda-vector-add.go:199]:
```
	if !ts.cfg.EKSConfig.IsEnabledAddOnCudaVectorAdd() {
		ts.cfg.Logger.Info("skipping tester.Delete", zap.String("tester", reflect.TypeOf(tester{}).PkgPath()))
		return nil
	}
```

File: eks/cuda-vector-add/cuda-vector-add.go [hunk 7]
```diff
@@ -0,0 +1,254 @@
...
+		ts.cfg.Logger.Info("skipping tester.AggregateResults", zap.String("tester", reflect.TypeOf(tester{}).PkgPath()))
+		return nil
+	}
+
+	ts.cfg.Logger.Info("starting tester.AggregateResults", zap.String("tester", reflect.TypeOf(tester{}).PkgPath()))
+	return nil
+}
+
+func (ts *tester) Create() error {
+	if ts.cfg.EKSConfig.AddOnCudaVectorAdd.Created {
```
Comment (gyuho) [inline eks/cuda-vector-add/cuda-vector-add.go:67]: Add

```
	if !ts.cfg.EKSConfig.IsEnabledAddOnCudaVectorAdd() {
		ts.cfg.Logger.Info("skipping tester.Create", zap.String("tester", reflect.TypeOf(tester{}).PkgPath()))
		return nil
	}
```

File: eks/cuda-vector-add/cuda-vector-add.go [hunk 8]
```diff
@@ -0,0 +1,254 @@
...
+func New(cfg Config) eks_tester.Tester {
+	cfg.Logger.Info("creating tester", zap.String("tester", reflect.TypeOf(tester{}).PkgPath()))
+	return &tester{cfg: cfg}
+}
+
+type tester struct {
+	cfg Config
+}
+
+func (ts *tester) AggregateResults() error {
```
Comment (gyuho) [inline eks/cuda-vector-add/cuda-vector-add.go:52]: Can we move `Aggregate...` to the bottom?

Comment (PatrickXYS) [reply eks/cuda-vector-add/cuda-vector-add.go]: Fixed.

File: eks/cuda-vector-add/cuda-vector-add.go (hunk 6 above)
Comment (PatrickXYS) [reply eks/cuda-vector-add/cuda-vector-add.go:185]: Right, we need to do this.

File: eks/cuda-vector-add/cuda-vector-add.go (hunk 7 above)
Comment (PatrickXYS) [reply eks/cuda-vector-add/cuda-vector-add.go]: Fixed.

**Pull request merged**